sections/collections__products-custom.liquid

```diff
@@ -1,10 +1,29 @@
 
+<!-- <div class="product-grid">
+    <ul class="display-grid">
+        {% for product in collection.products %}
+            <a href="{{ product.url }}">
+                <li class="product-grid__list-item">
+                    <img class="featured-image" src="{{ product.featured_image | img_url: 'medium' }}">
+                    {% for image in product.images %}
+                        {% if forloop.index0 == 1 %}
+                            <img class="hovered-image" src="{{ image | img_url: 'medium' }}">
+                        {% endif %}
+                    {% endfor %}
+                    <h4 class="product-grid__title">{{ product.title }}</h4>
+                    <h5 class="product-grid__vendor">{{ product.vendor }}</h5>
+                    <h6 class="product-grid__price">{{ product.price | money }}</h6>
+                </li>
+            </a>
+        {% endfor %}
+    </ul>
+</div> -->
 <div class="product-grid">
     <ul class="display-grid">
         {% for product in collection.products %}
             <a href="{{ product.url }}">
                 <li class="product-grid__list-item">
-                    <img src="{{ product.featured_image | img_url: 'medium' }}">
+                    <img class="featured-image" src="{{ product.featured_image | img_url: 'medium' }}" onmouseover="this.src='{% for image in product.images %}{% if forloop.index0 == 1 %}{{ image | img_url: 'medium' }}{% endif %}{% endfor %}'" onmouseout="this.src='{{ product.featured_image | img_url: 'medium' }}'"/>
```

Comment:
this is an interesting way of doing it. i've never seen it done like this, especially with just using`this.src=...`. 

The CSS way is also a good method/tool to keep in your back pocket. 

For the purposes of this website, since the image sizes are different, let's go with something similar to your initial solution but with a twist. Instead of using a second <img /> element (which downloads the image when the page is loaded), let's store the two image src's in a hidden input and modify your JS function (`$('.product-grid__list-item').each(function(){ ...` so that on hover, you update the <img> src with the second image, and on mouseout, update the src back to the first one. 

You can also do something similar to what you're currently doing, where you're setting the height after replacing the image. 

This way, we are meeting the criteria: replace image on hover, optimize for page load by not loading two images in two <img> elements. (The CSS way would also meet the criteria. That way would be the best solution if you had to lay text over the image or something like that)

Let me know if that's confusing.

Reply:
Got it. Updated.